### PR TITLE
added config functionality cli

### PR DESCRIPTION
**Pull request by siddhantrao23**
PR to solve issue #26 
Only problem is that the option in the click module doesn't allow variable length (2 args needed for get and remove but 3 args needed for set)

### CODE REVIEW

**Comment by CatEars** [comment]
I think you can use variadic argument length to not have to use `--config remove X -1` . See here: https://click.palletsprojects.com/en/7.x/arguments/

It shouldn't be too hard to parse since the first element is always the operation (set, get, remove) and the second element is always the key. For `set` there will be a third element. If there are more elements I think we can just discard those, so for example:

```
$ goto --config remove X Y Z
... Removed X ...
$
```

Will only print that X was removed. Should be enough UX for the user to understand Y and Z are left unchanged.

File: src/goto/storage.py [hunk 1]
```diff
@@ -207,7 +213,7 @@ def remove_teleport(name):
     '''Removes a teleport from the currently active profile.'''
     data = get_active_profile()
     if not data.get(name):
-        msg = '{} is not a location you can teleport to'.format(name)
+        msg = 'Error: {} is not a location you can teleport to'.format(name)
```
Comment (CatEars) [inline src/goto/storage.py]: I don't think it should say "Error:" In all error messages. That was probably unclear from my side in issue #26... Better to keep them like they are without "Error:" at the start.

File: src/goto/storage.py [hunk 2]
```diff
@@ -304,3 +310,26 @@ def get_prefix_expansions(prefix):
     subprefixes = list_subprefixes(prefix)
     teleport_no_prefix, _ = os.path.split(prefix)
     return [os.path.join(teleport_no_prefix, x) + os.sep for x in subprefixes]
+
+def set_config(attr, value):
+    '''sets attr to value'''
+    data = _get_settings()
+    data[attr] = int(value)
```
Comment (CatEars) [inline src/goto/storage.py]: Just putting in `value` directly is fine here. No need to convert to int. However, I do want a whitelist for this. So for example you can have a global `ALLOWED_SETTING_KEYS = set(['test_key'])` and check `if not attr in ALLOWED_SETTING_KEYS: ... raise StorageException(msg)`

Comment (siddhantrao23) [reply src/goto/storage.py]: I'll change these and get back asap.

**Comment by siddhantrao23** [comment]
Anything else you'd want me to add? I'm also interested in adding more functionality.

**Comment by CatEars** [comment]
I started a review for the current work. I think that if you fix those last parts we can merge this in. If you want to contribute some more then I think we can open another PR for that. I am interested in adding tests for `goto --config`, does that sound interesting?

Other than that if you are looking for more functionality there is the case of adding fuzzy matching for tab completion, for instance lets say I want to go to `/home/catears/xxx/yyy/zzz` and I have a teleport `x:/home/catears/xxx` then I would want to run `goto xyz` and it should figure out that I actually want to go to `xxx/yyy/zzz`. However there is a bit of issue here that you can implement this fuzzy matching in so many different ways that I wanted to have `--config` fixed for it, so you would get `--config FuzzyMatchingStyle A` or `--config FuzzyMatchingStyle B`. I have some more info on this in Issue #23.

**Comment by siddhantrao23** [comment]
I've fixed those last parts. Adding tests is something I've never done before but I'm willing to take a shot at it. I'll try that and get back to you, maybe through another PR?

**Comment by CatEars** [comment]
I see that I hadn't submitted the review, not sure if you could see the unfinished review. But those are the last things I want you to fix for this PR. If you think that fuzzy matching sounds interesting then I have described a bit more in the issue for it #23. I write a bit more there what I think is a good approach.

File: src/goto/cli.py [hunk 3]
```diff
@@ -185,12 +210,14 @@ def print_help():
...
     'profiles': 'List all profiles',
-    'install': 'Install goto for the given shell, "bash" or "zsh"'
+    'install': 'Install goto for the given shell, "bash" or "zsh"',
+    'config': 'Configure behaviour get/set/remove'
 }
 
 BASH_ZSH = click.Choice(['bash', 'zsh'])
 
 @click.command()
+@click.argument('configuration', nargs=-1)
```
[the remark this answers is not in the document]
Comment (siddhantrao23) [reply src/goto/cli.py]: nargs is a parameter for options but cannot be < 0, which we need in this case for variable length. This can only be done through vardiadic arguments, at least to my understanding.

[the remark this answers is not in the document]
Comment (siddhantrao23) [reply src/goto/cli.py]: Sorry about the many stupid issues! I'm still learning :P

[the remark this answers is not in the document]
Comment (CatEars) [reply src/goto/cli.py]: I made a quick click program and verified this, yeah. click doesn't allow -1 for `nargs`

```
import click

@click.command()
@click.option('--config', '-c', nargs=-1)
def main(**kwargs):
    print('Hello!', kwargs)

if __name__ == "__main__":
    main()
```

I think it would be nicer to have like --config-set, --config-get and --config-remove and we can specify exactly how many things and what types should be added but I think that can be done later down the line. For now this can use variadiac arguments, it's good enough!

I'll merge this into master! Congrats on being the first outside contributor to goto!

[the remark this answers is not in the document]
Comment (siddhantrao23) [reply src/goto/cli.py]: Thank you for your insights, it's been a pleasure. Excited to work on more for goto!
